Sources/ReactiveArray.swift

```diff
@@ -3,20 +3,25 @@ import ReactiveSwift
 import Result
 
 public final class ReactiveArray<Element> {
-
 	public typealias Snapshot = ContiguousArray<Element>
 	public typealias Change = Delta<Snapshot, IndexSet>
 
-	public let signal: Signal<Change, NoError>
+	fileprivate let storage: Storage<ContiguousArray<Element>>
```

Comment:
Note that the goal here is not making it completely thread safe. It is still _not_ thread safe. It just ensures that the producer synchronises with writes and deltas.